RecoLocalMuon/RPCRecHit/src/RPCPointProducer.cc

```diff
@@ -66,49 +65,64 @@ void RPCPointProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup
   if(incldt){
     edm::Handle<DTRecSegment4DCollection> all4DSegments;
     iEvent.getByToken(dt4DSegments, all4DSegments);
+      
+    if (MuonGeometryWatcher.check(iSetup)) TheDTObjectsMap_->fillObjectMapDT(iSetup);
+      
     if(all4DSegments.isValid()){
-      DTSegtoRPC DTClass(all4DSegments,iSetup,iEvent,debug,ExtrapolatedRegion);
+      DTSegtoRPC DTClass(all4DSegments,iSetup,iEvent, debug, ExtrapolatedRegion, TheDTObjectsMap_);
       std::auto_ptr<RPCRecHitCollection> TheDTPoints(DTClass.thePoints());     
       iEvent.put(TheDTPoints,"RPCDTExtrapolatedPoints"); 
     }else{
-      if(debug) std::cout<<"RPCHLT Invalid DTSegments collection"<<std::endl;
+      LogDebug("RPCPointProducer") <<  std::cout<<"RPCHLT Invalid DTSegments collection"<<std::endl;
     }
   }
 
   if(inclcsc){
     edm::Handle<CSCSegmentCollection> allCSCSegments;
     iEvent.getByToken(cscSegments, allCSCSegments);
+    
+      if (MuonGeometryWatcher.check(iSetup)) {
+          std::cout << "test  geom" << std::endl;
+          TheCSCObjectsMap_->fillObjectMapCSC(iSetup);
+      }
+      
     if(allCSCSegments.isValid()){
-      CSCSegtoRPC CSCClass(allCSCSegments,iSetup,iEvent,debug,ExtrapolatedRegion);
+      CSCSegtoRPC CSCClass(allCSCSegments,iSetup,iEvent, debug, ExtrapolatedRegion, TheCSCObjectsMap_);
       std::auto_ptr<RPCRecHitCollection> TheCSCPoints(CSCClass.thePoints());  
       iEvent.put(TheCSCPoints,"RPCCSCExtrapolatedPoints"); 
     }else{
-      if(debug) std::cout<<"RPCHLT Invalid CSCSegments collection"<<std::endl;
+      LogDebug("RPCPointProducer") << std::cout<<"RPCHLT Invalid CSCSegments collection"<<std::endl;
     }
   }
   if(incltrack){
     edm::Handle<reco::TrackCollection> alltracks;
     iEvent.getByToken(tracks,alltracks);
+      
+    if (MuonGeometryWatcher.check(iSetup)) {
+        TheDTtrackObjectsMap_->fillObjectMapDT(iSetup);
+        TheCSCtrackObjectsMap_->fillObjectMapCSC(iSetup);
+    }
+      
     if(!(alltracks->empty())){
-      TracktoRPC TrackClass(alltracks,iSetup,iEvent,debug,trackTransformerParam,tracks_);
+      TracktoRPC TrackClass(alltracks,iSetup,iEvent,trackTransformerParam,tracks_, TheDTtrackObjectsMap_, TheCSCtrackObjectsMap_);
       std::auto_ptr<RPCRecHitCollection> TheTrackPoints(TrackClass.thePoints());
       iEvent.put(TheTrackPoints,"RPCTrackExtrapolatedPoints");
     }else{
-      std::cout<<"RPCHLT Invalid Tracks collection"<<std::endl;
+      LogDebug("RPCPointProducer") <<"RPCHLT Invalid Tracks collection"<<std::endl;
     }
   }
  
 }
 
-// ------------ method called once each job just before starting event loop  ------------
-void 
-RPCPointProducer::beginJob()
-{
-}
+void  RPCPointProducer::beginRun(edm::Run const& iRun, edm::EventSetup const& iSetup){
+    TheDTObjectsMap_  = new ObjectMap(iSetup);
+    TheCSCObjectsMap_ = new ObjectMapCSC(iSetup);
+    TheDTtrackObjectsMap_ = new ObjectMap2(iSetup);
+    TheCSCtrackObjectsMap_ = new ObjectMap2CSC(iSetup);
```

Comment:
These four objects need to be deleted when they are no longer needed. I don't see anywhere where deletion occurs. You could put the deletion in the destructor or possibly in endRun().